parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -154,6 +154,14 @@ if [[ -n "${MASTER_NODE}" && "${KMS_PROVIDER_VAULT_NAME}" != "" ]]; then
     ensureKMS
 fi
 
+# configure and enable dhcpv6 for dual stack feature
+DHCPV6_SYSTEMD_SERVICE=/etc/systemd/system/dhcpv6.service
+DHCPV6_CONFIGURATION_SCRIPT=/opt/azure/containers/enable-dhcpv6.sh
+
+if [[ -f ${DHCPV6_SYSTEMD_SERVICE} ]] && [[ -f ${DHCPV6_CONFIGURATION_SCRIPT} ]];then
```

Comment:
There is a possible race condition here. If cloud-init fails to pave files before CustomScriptExtension code execution reaches this line, then we will skip this step, even if the cluster configuration declares that we *should* pave these files.

Reply:
Ooh yeah good catch... +1

Instead of checking if the file exist, you could add a variable for IS_IPV6 (with a better name) and use `wait_for_file` to wait for these two files to exist before preceding, similar to how we wait for the Azure Stack script https://github.com/Azure/aks-engine/blob/master/parts/k8s/cloud-init/artifacts/cse_main.sh#L34

Reply:
Done